### DISCUSSION TITLE
Volar with TS is complaining about QForm event datatype

### DISCUSSION THREAD

**Discussion by H25E** [quasarframework/quasar#12399] [General - Components / Directives / etc] [answered]
Hello,

I'm having trouble with QForm submit event and Volar TS data type checking:

	<template>
		<form @submit.prevent="sendLogin"></form>
		<q-form @submit.prevent="sendLogin"></q-form>
	</template>
	<script setup lang="ts">
		function sendLogin(e: Event) {return;}
	</script>

The `form` line it's ok but the `q-form` one says:

	Type '(e: Event) => void' is not assignable to type '($event?: undefined) => void'.
	Types of parameters 'e' and '$event' are incompatible.

It's a bug or a missconfiguration from my part?

I have found this [discussion](https://github.com/quasarframework/quasar/discussions/10619) but seems that is already implemented.

**Comment by rstoenescu** [comment, marked as answer]
It's SubmitEvent instead of Event.
Even though SubmitEvent is more accurate, I will add both to the definition in Quasar v2.5.5 so others won't stumble on this.

**Reply by H25E** [reply]
Thanks!!

I was using `SubmitEvent` at first, but with it the plain `<form>` was also complaining. Maybe it's a volar thing.
